api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2ClientConfig.java

```diff
@@ -234,61 +287,100 @@ void check() {
       }
-      checkArgument(
+      check(
+          violations,
+          CONF_NESSIE_OAUTH2_AUTHORIZATION_CODE_FLOW_TIMEOUT,
           getAuthorizationCodeFlowTimeout().compareTo(getMinAuthorizationCodeFlowTimeout()) >= 0,
           "authorization code flow: timeout must be greater than or equal to %s",
-          Duration.ofSeconds(10));
+          getMinAuthorizationCodeFlowTimeout());
     }
     if (grantType == GrantType.DEVICE_CODE) {
-      checkArgument(
+      check(
+          violations,
+          CONF_NESSIE_OAUTH2_ISSUER_URL + "/" + CONF_NESSIE_OAUTH2_AUTH_ENDPOINT,
           getIssuerUrl().isPresent() || getDeviceAuthEndpoint().isPresent(),
           "either issuer URL or device authorization endpoint must be set if grant type is '%s'",
           CONF_NESSIE_OAUTH2_GRANT_TYPE_DEVICE_CODE);
-      checkArgument(
+      check(
+          violations,
+          CONF_NESSIE_OAUTH2_DEVICE_CODE_FLOW_POLL_INTERVAL,
           getDeviceCodeFlowPollInterval().compareTo(getMinDeviceCodeFlowPollInterval()) >= 0,
           "device code flow: poll interval must be greater than or equal to %s",
-          Duration.ofSeconds(5));
-      checkArgument(
+          getMinDeviceCodeFlowPollInterval());
+      check(
+          violations,
+          CONF_NESSIE_OAUTH2_DEVICE_CODE_FLOW_TIMEOUT,
           getDeviceCodeFlowTimeout().compareTo(getMinDeviceCodeFlowTimeout()) >= 0,
           "device code flow: timeout must be greater than or equal to %s",
-          Duration.ofSeconds(10));
+          getMinDeviceCodeFlowTimeout());
     }
-    checkArgument(
+    check(
+        violations,
+        CONF_NESSIE_OAUTH2_DEFAULT_ACCESS_TOKEN_LIFESPAN,
         getDefaultAccessTokenLifespan().compareTo(getMinDefaultAccessTokenLifespan()) >= 0,
         "default token lifespan must be greater than or equal to %s",
         getMinDefaultAccessTokenLifespan());
-    checkArgument(
+    check(
+        violations,
+        CONF_NESSIE_OAUTH2_REFRESH_SAFETY_WINDOW,
         getRefreshSafetyWindow().compareTo(getMinRefreshSafetyWindow()) >= 0,
         "refresh safety window must be greater than or equal to %s",
         getMinRefreshSafetyWindow());
-    checkArgument(
+    check(
+        violations,
+        CONF_NESSIE_OAUTH2_REFRESH_SAFETY_WINDOW
+            + "/"
+            + CONF_NESSIE_OAUTH2_DEFAULT_ACCESS_TOKEN_LIFESPAN,
         getRefreshSafetyWindow().compareTo(getDefaultAccessTokenLifespan()) < 0,
         "refresh safety window must be less than the default token lifespan");
-    checkArgument(
+    check(
+        violations,
+        CONF_NESSIE_OAUTH2_PREEMPTIVE_TOKEN_REFRESH_IDLE_TIMEOUT,
         getPreemptiveTokenRefreshIdleTimeout().compareTo(getMinPreemptiveTokenRefreshIdleTimeout())
             >= 0,
         "preemptive token refresh idle timeout must be greater than or equal to %s",
         getMinPreemptiveTokenRefreshIdleTimeout());
+
+    // This check is rather useless, because the executor is created
```

Comment:
not sure I understand this comment, sorry :thinking:

Reply:
When I started to write it, I realized that it's not good to have the executor in this type, because that's created before the config's validated.